### ISSUE TITLE
[Potential NAN bug] Loss may become NAN during training

### ISSUE THREAD

**Issue by Justobe** [jxyyjm/tensorflow_test#3]
Hello~

Thank you very much for sharing the code!

I try to use my own data set ( with the same shape as mnist) in  code. After some iterations, it is found that the training loss becomes NAN. After carefully checking the code, I found that the following code may trigger NAN in loss:

In `tensorflow_test/src/simple_saveAndReloadModel.py`

```python
cross_entropy = -tf.reduce_sum(y_ * tf.log(y))
```

If  y contains 0 (output of softmax ), the result of `tf.log(y)` is `inf` because `log(0)` is illegal . And this may cause the result of loss to become NAN.

It could be fixed by making the following changes:

```python
cross_entropy = -tf.reduce_sum(y_ * tf.log(y + 1e-8))
```

or

```python
cross_entropy = -tf.reduce_sum(y_ * tf.log(tf.clip_by_value(y,1e-8,1.0)))
```

Hope to hear from you ~

Thanks in advance!  : )
